patches/0004-desktop-editor-intigration.patch

```diff
@@ -0,0 +1,237 @@
+-      return spawn(foundShell.path, ['--working-directory', path])
+-    case Shell.Urxvt:
+-      return spawn(foundShell.path, ['-cd', path])
+-    case Shell.Konsole:
+-      return spawn(foundShell.path, ['--workdir', path])
+-    case Shell.Xterm:
+-      return spawn(foundShell.path, ['-e', '/bin/bash'], { cwd: path })
+-    case Shell.Terminology:
+-      return spawn(foundShell.path, ['-d', path])
+-    case Shell.Deepin:
+-      return spawn(foundShell.path, ['-w', path])
+-    case Shell.Elementary:
+-      return spawn(foundShell.path, ['-w', path])
+-    case Shell.Kitty:
+-      return spawn(foundShell.path, ['--single-instance', '--directory', path])
+-    default:
+-      return assertNever(shell, `Unknown shell: ${shell}`)
++  if (process.env.FLATPAK_HOST !== 'undefined') {
++    switch (shell) {
++      case Shell.Gnome:
++      case Shell.Mate:
++      case Shell.Tilix:
++      case Shell.Terminator:
++      case Shell.XFCE:
++      case Shell.Alacritty:
++        return spawn("flatpak-spawn", ["--host", foundShell.path, '--working-directory', path])
++      case Shell.Urxvt:
++        return spawn("flatpak-spawn", ["--host", foundShell.path, '-cd', path])
++      case Shell.Konsole:
++        return spawn("flatpak-spawn", ["--host", foundShell.path, '--workdir', path])
++      case Shell.Xterm:
++        return spawn("flatpak-spawn", ["--host", foundShell.path, '-e', '/bin/bash'], { cwd: path })
++      case Shell.Terminology:
++        return spawn("flatpak-spawn", ["--host", foundShell.path, '-d', path])
++      case Shell.Deepin:
++        return spawn("flatpak-spawn", ["--host", foundShell.path, '-w', path])
++      case Shell.Elementary:
++        return spawn("flatpak-spawn", ["--host", foundShell.path, '-w', path])
++      case Shell.Kitty:
++        return spawn("flatpak-spawn", ["--host", foundShell.path, '--single-instance', '--directory', path])
++      default:
++        return assertNever(shell, `Unknown shell: ${shell}`)
++    }
++  } else {
++    switch (shell) {
++      case Shell.Gnome:
++      case Shell.Mate:
++      case Shell.Tilix:
++      case Shell.Terminator:
++      case Shell.XFCE:
++      case Shell.Alacritty:
++        return spawn(foundShell.path, ['--working-directory', path])
++      case Shell.Urxvt:
++        return spawn(foundShell.path, ['-cd', path])
++      case Shell.Konsole:
++        return spawn(foundShell.path, ['--workdir', path])
++      case Shell.Xterm:
++        return spawn(foundShell.path, ['-e', '/bin/bash'], { cwd: path })
++      case Shell.Terminology:
++        return spawn(foundShell.path, ['-d', path])
++      case Shell.Deepin:
++        return spawn(foundShell.path, ['-w', path])
++      case Shell.Elementary:
++        return spawn(foundShell.path, ['-w', path])
++      case Shell.Kitty:
++        return spawn(foundShell.path, ['--single-instance', '--directory', path])
++      default:
++        return assertNever(shell, `Unknown shell: ${shell}`)
++    }
```

Comment:
Maybe we could do something like this to not repeat ourselves:

```typescript
    var shellArgs = null, spawnArgs = {}
    switch (shell) {
      case Shell.Gnome:
      case Shell.Mate:
      case Shell.Tilix:
      case Shell.Terminator:
      case Shell.XFCE:
      case Shell.Alacritty:
        shellArgs = ['--working-directory', path]
        break
      case Shell.Urxvt:
        shellArgs = ['-cd', path]
        break
      case Shell.Konsole:
        shellArgs = ['--workdir', path]
      case Shell.Xterm:
        shellArgs = ['-e', '/bin/bash']
        spawnArgs = { cwd: path }
        break
      case Shell.Terminology:
        shellArgs = ['-d', path]
        break
      case Shell.Deepin:
      case Shell.Elementary:
        shellArgs = ['-w', path]
        break
      case Shell.Kitty:
        shellArgs = ['--single-instance', '--directory', path]
      default:
        return assertNever(shell, `Unknown shell: ${shell}`)
    }

    if (process.env.FLATPAK_HOST !== 'undefined') {
        return spawn("flatpak-spawn", ["--host", foundShell.path, ...shellArgs], spawnArgs)
    } else {
        return spawn(foundShell.path, shellArgs, spawnArgs)
    }
```

Reply:
I'm working on an upstream PR with @advaithm to make this patch easier to support in the main project (and eventually redundant) https://github.com/shiftkey/desktop/pull/555

Reply:
this pr has been merged, up stream should have a new release soon. I'm marking this as resolved